kubernetes/apps/external-secrets.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: external-secrets
+spec:
+  goTemplate: true
+  generators:
+    # targets all clusters
+    - clusters:
+        selector:
+          matchExpressions:
+            - key: clusterType
+              operator: Exists
+  template:
+    metadata:
+      name: 'external-secrets-{{ .name }}'
+    spec:
+      destination:
+        namespace: external-secrets
+        server: "{{ .server }}"
+      project: default
+      sources:
+        - chart: external-secrets
+          repoURL: 'https://charts.external-secrets.io'
+          targetRevision: v0.9.18
+          helm:
+            releaseName: external-secrets
+            parameters:
+              - name: installCRDs
+                value: 'true'
+              - name: serviceAccount.name
+                value: external-secrets
+            valueFiles:
+            - $values/kubernetes/{{ .name }}/helm/external-secrets.yaml
```

Comment:
How does this `$values` resolve? Is it resolving to the source defined below?

Reply:
It's resolved from the values defined below.